test/regression/sql/time_bucket.sql

```diff
@@ -0,0 +1,255 @@
+-- All functions should automatically use duckdb
+set duckdb.execution = false;
+
+CREATE TABLE dates(w INTERVAL, d DATE, shift INTERVAL, origin DATE);
+
+INSERT INTO dates VALUES
+	('10 days', '0794-11-15', '1 week', '0790-11-01'),
+	('10 days', '1700-01-01', '0 days', '1700-01-05'),
+	('1 week', '1832-05-03', '0 days', '1970-05-05'),
+	('10 days', '1897-07-05', '2 days', '1970-06-07'),
+	('2 months', '1946-09-14', '0 months', '1970-07-05'),
+	('2 months', '2000-01-01', '1 month 1 week', '1970-05-01'),
+	('2 year', '2004-05-20', '6 months', '1970-12-31');
+
+select d, time_bucket('3 days'::interval, d) from dates;
```

Comment:
While extensive tests are usually good. I think the amount of tests here is a bit excessive. We don't really need to ensure that the `time_bucket` function returns correct results for all weird inputs. DuckDB already tests for that. All we really need to test is that you can use this function from Postgres. So I think the only tests that we need is:

1. A single test for each of the variants of the function (so one per combination of arguments).
2. A test where the input to the function comes from `duckdb.query`, so that we test the `unresolved_type` support.